src/webui/abstractwebapplication.cpp

```diff
@@ -189,42 +197,28 @@ bool AbstractWebApplication::sessionInitialize()
 
 bool AbstractWebApplication::readFile(const QString& path, QByteArray &data, QString &type)
 {
-    QString ext = "";
-    int index = path.lastIndexOf('.') + 1;
-    if (index > 0)
-        ext = path.mid(index);
-
-    // find translated file in cache
-    if (translatedFiles_.contains(path)) {
-        data = translatedFiles_[path];
+    QFile file(path);
+    if (!file.open(QIODevice::ReadOnly)) {
+        qDebug("File %s was not found!", qUtf8Printable(path));
+        return false;
     }
-    else {
-        QFile file(path);
-        if (!file.open(QIODevice::ReadOnly)) {
-            qDebug("File %s was not found!", qUtf8Printable(path));
-            return false;
-        }
-
-        data = file.readAll();
-        file.close();
+    data = file.readAll();
+    file.close();
 
-        // Translate the file
-        if ((ext == "html") || ((ext == "js") && !path.endsWith("excanvas-compressed.js"))) {
-            QString dataStr = QString::fromUtf8(data.constData());
-            translateDocument(dataStr);
+    type = getContentType(path);
 
-            if (path.endsWith("about.html") || path.endsWith("index.html") || path.endsWith("client.js"))
-                dataStr.replace("${VERSION}", QBT_VERSION);
-
-            data = dataStr.toUtf8();
-            translatedFiles_[path] = data; // cashing translated file
-        }
-    }
-
-    type = CONTENT_TYPE_BY_EXT[ext];
     return true;
 }
 
+QString AbstractWebApplication::getContentType(const QString &path)
+{
+    QString ext = "";
```

Comment:
the assignment is redundant.

Reply:
and should be possible to use `QStringRef` if you like.

Reply:
Ok.

Reply:
>and should be possible to use QStringRef if you like.

Don't see any sense. It will be converted to QString later.

Reply:
>It will be converted to QString later.

OK and I'll continue the review later.